live/js.go

```diff
@@ -0,0 +1,198 @@
+package live
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// DefaultTransitionDuration_ms is the default duration, in milliseconds, of live.JS transitions.
+const DefaultTransitionDuration_ms = int32(200)
+
+type op struct {
+	kind string
+	args map[string]any
+}
+
+// JS provides a way to precompose simple client-side DOM changes that don't require a round-trip to the server.
+// Create a JS struct, call its methods to build up a command, and then render it as the value of a phx-* attribute.
+type JS struct {
+	ops []*op
+}
+
+// Hide hides elements.
+// When the action is triggered on the client, phx:hide-start is dispatched to the hidden elements. After the time specified by :time, phx:hide-end is dispatched.
+func (js *JS) Hide(opts *HideOpts) {
```

Comment:
Any reason not to make these "Fluent Interface" style?
```go
func (js *JS) Show(opts *ShowOpts) *JS {
	js.add("show", opts)
	return js
}
```
Then you can chain a bunch if need be (particularly inside a template):
```go
js.Show(...).Push(...)
```

Reply:
Oh right.  Go templates don't make calling these very easy at all...  No need to add rn.

Reply:
@floodfx Yeah, the entire motivation to support methods is that at present `JS` isn't intended to be called inside a template at all—originally I had everything variadic to support it, but the optional params make it tough. Easier to wrap it in view methods and call those. The plan is to see what we/others do in terms of wrapping `JS` in funcmaps, then perhaps implement a good version of those wrappers.